mydumper/mydumper.go

```diff
@@ -152,6 +152,11 @@ func (m *Mydumper) spawn(ctx context.Context) ([]byte, error) {
 	// so we parse all these lines and translate into our own logs.
 	scanner := bufio.NewScanner(stderrPipe)
 	for scanner.Scan() {
+		// handle errors in scanning
+		if err != nil {
+			stdout.Write(stderr.Bytes())
+			return stdout.Bytes(), err
```

Comment:
is it possible to lose some remaining log? 😂

Reply:
change this error-check to where after flush line, and correct return output message

Reply:
seems need to drain and output `stderr` of `mydumper` 😂

Reply:
when `terror.ErrDumpUnitGlobalLock` hanppens, there could be some remaining log which mydumper has output to `stderr`. Since go doesn't have a non-blocking IO read, I guess it's not neat to fetch these remaining log. So if we don't output remaining log, behaviour could be just like "stop on first error", is this acceptable?

Reply:
* non-blocking IO could be implement by another goroutine and select, not very hard though

Reply:
Can we just:

1. record the first error
2. out all log as we did before
3. return the recorded error

Reply:
OK, then we depend on mydumper will exit quickly so we could get the error quickly. I'll left a TODO to switch to non-blocking read If we want to detect more error and when that error happens, mydumper continue to run.

Reply:
done in latest two commit